### DISCUSSION TITLE
Custom entry on vulnerability analysis dropdown menu

### DISCUSSION THREAD

**Discussion by fabiolanza** [DependencyTrack/dependency-track#2179] [Q&A]
Is it possible to add a custom entry in the vulnerability analysis dropdown menu?

**Comment by stevespringett** [comment]
Not currently. What specifically are you looking for?

**Reply by fabiolanza** [reply]
Thanks for answering. I don't have a precise answer now, but it would be good to have the flexibility to add custom entries in order to accomodate any particularities in the future.

**Comment by fabiolanza** [comment]
Hi @stevespringett, in order to manage the analysis scenario where we downgrade a CVE severity due to our Intended Operational Environment, we'd like to experiment with the following entries in the vulnerability analysis dropdown menu:
- ECVES Critical
- ECVES High
- ECVES Medium
- ECVES Low

ECVES meaning Environmental CVE Severity.

Is it possible to have these entries added in that menu? How do we do that in our Dependency-Track instance and how to revert if the test outcomes are not good?

Thanks,
Fabio

**Reply by stevespringett** [reply]
We cannot alter the vulnerability analysis drop down too much. The values need to align to the CycloneDX standard. See https://cyclonedx.org/docs/1.4/json/#vulnerabilities_items_analysis_state

However, CycloneDX does support multiple ratings, each rating can be from a different source (NVD vs Your Company) and may be CVSS or OWASP risk rating. So ideally, there would be a way to apply an alternative rating for the vulnerability in DT.
See https://cyclonedx.org/docs/1.4/json/#vulnerabilities_items_ratings_items_method
